### PR TITLE
Azure app service app template

### PR DESCRIPTION
**Pull request by likelymichael**
Application template for Azure App Service. Module here: https://github.com/massdriver-cloud/terraform-modules/pull/33

closes: https://github.com/massdriver-cloud/massdriver/issues/1304

### CODE REVIEW

File: azure-app-service/massdriver.yaml [hunk 1]
```diff
@@ -0,0 +1,195 @@
...
+          type: integer
+          minimum: 1
+          maximum: 30
+        maximum_worker_count:
+          title: Maximum instances
+          description: Maximum number of instances for autoscale.
+          type: integer
+          minimum: 1
+          maximum: 30
+        notification_email:
```
Comment (chrisghill) [inline azure-app-service/massdriver.yaml]: FYI, we actually send down the email of the contact that owns the target(?) as part of md_metadata now. `var.md_metadata.target.contact_email`.  We can ask them here, or we can just use that e-mail to simplify things.

File: azure-app-service/massdriver.yaml [hunk 2]
```diff
@@ -0,0 +1,195 @@
...
+          title: Notification email
+          description: Receive alerts regarding autoscale events.
+          type: string
+          format: email
+          example: user@example.com
+    repo:
+      title: Repository
```
Comment (chrisghill) [inline azure-app-service/massdriver.yaml]: Let's make it clear this is the Docker images settings.  Here are the settings in the `kubernetes-deployment` app: https://github.com/massdriver-cloud/application-templates/blob/main/kubernetes-deployment/massdriver.yaml#L24-L41

File: azure-app-service/massdriver.yaml (hunk 1 above)
Comment (likelymichael) [reply azure-app-service/massdriver.yaml]: Would it be the same variable structure with this application template? For example, in the application module, I was encouraged to rename `var.md_metadata.name_prefix` to just `var.name`

File: azure-app-service/massdriver.yaml [hunk 3]
```diff
@@ -0,0 +1,195 @@
...
+          title: Enable DNS
+          description: Enable DNS configuration for the application.
+          type: boolean
+      dependencies:
+        enable_dns:
+          oneOf:
+            - properties:
+                enable_dns:
+                  const: true
+                zone_name:
```
Comment (chrisghill) [inline azure-app-service/massdriver.yaml:126]: Can we simplify these settings at all?  People who don't know DNS much will simply know they want their app on a certain URL. They may not understand the different between TXT records and CNAME records.  We can make the CNAME and generate a TXT record for validation for them.

Comment (likelymichael) [reply azure-app-service/massdriver.yaml:126]: Yea we can, I'll work on this tonight

File: azure-app-service/massdriver.yaml (hunk 1 above)
Comment (likelymichael) [reply azure-app-service/massdriver.yaml]: Removed from here and updated on the application module. Though I'm not sure how it gets passed through in the module itself.

File: azure-app-service/massdriver.yaml (hunk 2 above)
Comment (likelymichael) [reply azure-app-service/massdriver.yaml]: Updating this block to be similar to linked suggestion.

File: azure-app-service/massdriver.yaml [hunk 4]
```diff
@@ -0,0 +1,181 @@
...
+    - __name: Development
+      application:
+        sku_name: S1
+        minimum_worker_count: 1
+        maximum_worker_count: 3
+    - __name: Production
+      application:
+        sku_name: P3v3
+        minimum_worker_count: 1
+        maximum_worker_count: 10
```
Comment (jaketf) [inline azure-app-service/massdriver.yaml]: I had a similar conversation w/ beebe on cloud run. 10 feels really low for a production service that may need to scale. We don't want people to look at this guided config and think "massdriver only supports toy-scale apps" app service default max workers is 30 I say we at least follow that.

From azure best practices:
>The default limit of App Service workers is 30. If the App Service routinely uses 15 or more instances, consider opening a support ticket to increase the maximum number of workers to 2x the instance count required to serve normal peak load.

Comment (jaketf) [reply azure-app-service/massdriver.yaml]: https://docs.microsoft.com/en-us/azure/architecture/framework/services/compute/azure-app-service/operational-excellence

File: azure-app-service/massdriver.yaml [hunk 5]
```diff
@@ -0,0 +1,181 @@
...
+  examples:
+    - __name: Development
+      application:
+        sku_name: S1
+        minimum_worker_count: 1
+        maximum_worker_count: 3
+    - __name: Production
+      application:
+        sku_name: P3v3
+        minimum_worker_count: 1
```
Comment (jaketf) [inline azure-app-service/massdriver.yaml]:
```suggestion
        minimum_worker_count: 2
```
azure suggests

> Use Basic or higher plans with two or more worker instances for high availability.

https://docs.microsoft.com/en-us/azure/architecture/framework/services/compute/azure-app-service/operational-excellence

Comment (likelymichael) [reply azure-app-service/massdriver.yaml]: The customer will be paying for those additional instances. If they want to spin up an app for dev/test, I don't think they'll want 2 instances running at double the cost. Under the guided config for app templates, I can set the recommended minimum worker count for production environments.

File: azure-app-service/massdriver.yaml [hunk 6]
```diff
@@ -0,0 +1,181 @@
...
+      title: Container Image
+      type: object
+      required:
+      - repository
+      - tag
+      properties:
+        repository:
+          $md.immutable: true
+          title: Repository
+          # Massdriver automatically configures your Kubernetes clusters to have access to your cloud container repositories.
```
Comment (coryodaniel) [inline azure-app-service/massdriver.yaml]: Swap kubernetes for app service

**Review by coryodaniel — APPROVED** [review]

File: azure-app-service/massdriver.yaml (hunk 5 above)
Comment (jaketf) [reply azure-app-service/massdriver.yaml]: yeah i agree. this comment was just to change it in the production guided config

Comment (likelymichael) [reply azure-app-service/massdriver.yaml]: Oops didn't notice that. Thanks!

**Pull request merged**